### PR TITLE
Adding Olive-network contracts for TVL computation

### PR DESCRIPTION
**Pull request by nisuscold**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/olive-network exports TVL: 
        
 
 
    ethereum                  4.62 M
    arbitrum                  2.61 M
    
    total                    7.23 M

File: projects/olive-network/index.js
Comment (g1nt0ki) [inline projects/olive-network/index.js:1]: hi, thanks for the PR, writing it like below is bit more efficient & easier to read:

    const { sumERC4626Vaults } = require('../helper/erc4626')
    const config = {
      ethereum: {
        vaults: [
          // Token vaults
          "0x700886a402d42113aD94D9756f08A923BB5aC77A", // eETH
          "0xDa4D36dbdf5154B22580c0f1c998D37BfBB33D85", // ezETH
          "0xd75c669B3da058cf589bF0076FDaceDa40380C4d", // pufETH
          "0x9F3E781b25501A6b9051556B8058812D7Ba30549", // rsETH
          "0x6De2a95331400bb6cf9Cf75c7a8861d33687a95F", // stETH
          "0x7531b2AbA509E09566C08D61CaD6324b78444eCd", // wBTC
          "0xdb0d6F58a63118E20C91c0De84f4d8eA1a407C36", // weETH
          "0x2865568AD1CA0FE12dB53c8f866039Fa4500962F", // wETH
          "0xDEA5f3171C5052384a0a974E3C85b0d419c48204", // wstETH

          // PT vaults
          "0xeF0DF466417bC45007773C363866B3693fc0b1E7", // PT-weETH Jun 2024
          "0x31ec45f7dA20998775d594539F54e443e268f9F5", // PT-ezETH Apr 2024
          "0xF3b442217f18EB46417eFfd3A6cE09C3B311f4f5", // PT-pufETH Jun 2024
          "0x6D8c42855690c493E9c6404803478CD321A63376", // PT-rsETH Jun 2024
          "0x47764d88F8f54daD6Db75EC3667a11d58811ABc6", // PT-wstETH Dec 2024
        ], ethVaults: ['0xd94a9FBae86e662350FFEbB352f70c3CBeb9E96e'],
      },
      arbitrum: {
        vaults: [
          // PT vaults
          "0xA2e9c6177f81a1337b656d3b066FD20c7Cf8cAb4", // ezETH
          "0xd94a9FBae86e662350FFEbB352f70c3CBeb9E96e", // rsETH
          "0x2865568AD1CA0FE12dB53c8f866039Fa4500962F", // weETH
          "0x6De2a95331400bb6cf9Cf75c7a8861d33687a95F", // wstETH
          "0x700886a402d42113aD94D9756f08A923BB5aC77A", // PT-weETH Jun 2024
          "0xDa4D36dbdf5154B22580c0f1c998D37BfBB33D85", // PT-ezETH Jun 2024
          "0xd75c669B3da058cf589bF0076FDaceDa40380C4d", // PT-rsETH Jun 2024
          "0x9F3E781b25501A6b9051556B8058812D7Ba30549", // PT-wstETH Dec 2024
        ],
      },
    }

    Object.keys(config).forEach(chain => {
      const { vaults, ethVaults = [], } = config[chain]
      module.exports[chain] = {
        tvl: async (api) => {
          const ethBals = await api.multiCall({ abi: 'uint256:balance', calls: ethVaults })
          api.addGasToken(ethBals)
          return sumERC4626Vaults({ api, calls: vaults })
        }
      }
    })

**Comment by llamatester** [comment]
The adapter at projects/olive-network exports TVL: 
        
 
 
    ethereum                  2.56 M
    arbitrum                  2.54 M
    
    total                    5.10 M

**Comment by llamatester** [comment]
The adapter at projects/olive-network exports TVL: 
        
 
 
    ethereum                  2.57 M
    arbitrum                  2.54 M
    
    total                    5.11 M

File: projects/olive-network/index.js
Comment (nisuscold) [reply projects/olive-network/index.js:1]: Thanks for the suggestion, implemented the same.

**Comment by llamatester** [comment]
The adapter at projects/olive-network exports TVL: 
        
 
 
    ethereum                  2.57 M
    arbitrum                  2.54 M
    
    total                    5.11 M

**Comment by g1nt0ki** [comment]
is it ready for review?

**Comment by llamatester** [comment]
The adapter at projects/olive-network exports TVL: 
        
 
 
    ethereum                  2.60 M
    arbitrum                  2.59 M
    
    total                    5.19 M

**Comment by nisuscold** [comment]
> is it ready for review?

Yes the PR is ready for review.

**Comment by llamatester** [comment]
The adapter at projects/olive-network exports TVL: 
        
 
 
    ethereum                  2.57 M
    arbitrum                  2.56 M
    
    total                    5.14 M

**Comment by g1nt0ki** [comment]
@nisuscold thanks for the PR

**Pull request merged**